lib/ramble/ramble/application.py

```diff
@@ -557,12 +572,19 @@ def build_used_variables(self, workspace):
             elif criteria.mode == "application_function":
                 self.evaluate_success()
 
+        if self.package_manager is not None:
+            self.package_manager.build_used_variables(workspace)
+
         for template_name, template_conf in workspace.all_templates():
             self.expander._used_variables.add(template_name)
             self.expander.expand_var(template_conf["contents"])
 
-        if self.package_manager is not None:
-            self.package_manager.build_used_variables(workspace)
+        ############################
+        # Reset variable definitions
+        ############################
+
+        for var, val in backup_variables.items():
```

Comment:
Does that mean we are OK with keeping new variables added?

Reply:
Good point. I meant to delete the difference, so let me fix that up